File: backend/app/schemas/info_request.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from app.models.info_request import RequestStatus

class InfoRequestBase(BaseModel):
    lead_id: int
    field_name: str
    note: Optional[str] = None

class InfoRequestCreate(InfoRequestBase):
    pass

class InfoRequestUpdate(BaseModel):
    status: RequestStatus
    assigned_to: Optional[int] = None
    note: Optional[str] = None

class InfoRequestResponse(InfoRequestBase):
    id: int
    status: RequestStatus
    requested_by: int
    assigned_to: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/infoRequest.ts
export type RequestStatus = 'PENDING' | 'IN_PROGRESS' | 'COMPLETED' | 'REJECTED';

export interface InfoRequest {
  id: number;
  lead_id: number;
  field_name: string;
  note?: string;
  status: RequestStatus;
  requested_by: number;
  assigned_to?: number;
  created_at: string;
  updated_at?: string;
  completed_at?: string;
}

export interface InfoRequestCreate {
  lead_id: number;
  field_name: string;
  note?: string;
}

export interface InfoRequestUpdate {
  status: RequestStatus;
  assigned_to?: number;
  note?: string;
}
